zcash_client_sqlite/src/lib.rs

```diff
@@ -1425,11 +1485,49 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                                 txout.value,
                                 None,
                             )?;
+
+                            // Even though we know the funding account, we don't know that we have
+                            // information for all of the transparent inputs to the transaction.
+                            #[cfg(feature = "transparent-inputs")]
+                            {
+                                tx_has_wallet_outputs = true;
+                            }
+
+                            // If the decrypted transaction is unmined and has no shielded
+                            // components, add it to the queue for status retrieval.
+                            #[cfg(feature = "transparent-inputs")]
+                            if d_tx.mined_height().is_none() && !detectable_via_scanning {
+                                wallet::queue_tx_retrieval(
+                                    wdb.conn.0,
+                                    std::iter::once(d_tx.tx().txid()),
+                                    TxQueryType::Status,
+                                    None
+                                )?;
+                            }
                         }
                     }
                 }
             }
 
+            // If the transaction has outputs that belong to the wallet as well as transparent
+            // inputs, we may need to download the transactions corresponding to the transparent
+            // prevout references to determine whether the transaction was created (at least in
+            // part) by this wallet.
+            #[cfg(feature = "transparent-inputs")]
+            if tx_has_wallet_outputs {
+                if let Some(b) = d_tx.tx().transparent_bundle() {
+                    // queue the transparent inputs for enhancement
+                    wallet::queue_tx_retrieval(
+                        wdb.conn.0,
+                        b.vin.iter().map(|txin| *txin.prevout.txid()),
+                        TxQueryType::Enhancement,
+                        Some(tx_ref)
+                    )?;
+                }
+            }
+
+            notify_tx_retrieved(wdb.conn.0, d_tx.tx().txid())?;
```

Comment:
Since we're in a database transaction, I think [this problem](https://github.com/zcash/librustzcash/pull/1473/files#r1711669805) can probably be solved just by moving this call to be earlier.